File: todo-java/src/main/java/com/renan/todo/dto/UserDTO.java
package com.renan.todo.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * User DTO - with de token
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {

    private String email;
    private String jwt;
    private Integer id;
    private Integer status;
    private Integer expiresIn;
    private String name;
    private String lastName;

}

File: to-do app/to-do-go/src/controller/response/user_dto.go
package response

import "renan.com/todo/src/security"

// UserDTO struct with data to use as response for login operations
type UserDTO struct {
	Email     string `json:"email"`
	JWT       string `json:"jwt"`
	ID        uint64 `json:"id"`
	Status    uint64 `json:"status"`
	ExpiresIn uint64 `json:"expiresIn"`
	Name      string `json:"name"`
	LastName  string `json:"lastName"`
}

//CreateUserDTOFromJWT create a new DTO Object from the token
func CreateUserDTOFromJWT(jwtToken string) UserDTO {
	userID, status, email, firstName, lastName, err := security.GetUserDataFromToken(jwtToken)
	if err != nil {
		return UserDTO{}
	}
	newToken, err := security.CreateToken(userID, status, email, firstName, lastName)
	if err != nil {
		return UserDTO{}
	}
	return UserDTO{
		Email:     email,
		ID:        userID,
		Status:    status,
		Name:      firstName,
		LastName:  lastName,
		ExpiresIn: uint64(security.EXP_TIME),
		JWT:       newToken,
	}
}
